### PR TITLE
EOS-24154: Handle non-existent component support bundle generation

### PR DESCRIPTION
**Pull request by vastradparayya**
Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

# Problem Statement
- Handle non-existent component support bundle generation
- User should be get to know when he entered wrong component in SB generation

# Design
-  https://jts.seagate.com/browse/EOS-24154

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: Y
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: Y
- [x] Confirm Testing was performed with installed RPM [Y/N]:  Y

# Review Checklist 
  Before posting the PR please ensure
- [x ] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: N
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/test/support_bundle/test_support_bundle.py [hunk 1]
```diff
@@ -65,7 +65,8 @@ def test_004status(self):
         status = SupportBundle.get_status(bundle_id=bundle_obj.bundle_id)
         self.assertIsNotNone(status)
         status = json.loads(status)
-        self.assertEqual(status['status'][0]['result'], "Success")
+        if status['status']:
+            self.assertEqual(status['status'][0]['result'], "Success")
```
Comment (selvakumaar5496) [inline py-utils/test/support_bundle/test_support_bundle.py:69]: single quote on Success

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 2]
```diff
@@ -110,6 +110,19 @@ async def _generate_bundle(command):
...
+        
+        support_bundle_file = os.path.join(Conf.get('cortx_conf', 'install_path'),\
+            'cortx/utils/conf/support_bundle.yaml')
+        Conf.load('sb_yaml', f'yaml://{support_bundle_file}', skip_reload=True)
+        all_components = Conf.get('sb_yaml', 'COMPONENTS')
+        invalid_comps = [component for component in components if component not in all_components.keys()]
+        if invalid_comps:
+            components = list(set(components) - set(invalid_comps))
+            ComponentsBundle._publish_log(f"""Invalid components - '{", ".join(invalid_comps)}'""", \
+                'error', bundle_id,'', comment)
```
Comment (selvakumaar5496) [inline py-utils/src/utils/support_framework/support_bundle.py:122]: space after comma

**Review by rahul-tripathi-git — APPROVED** [review]
Looks good to me.

File: py-utils/test/support_bundle/test_support_bundle.py (hunk 1 above)
Comment (vastradparayya) [reply py-utils/test/support_bundle/test_support_bundle.py]: Done

File: py-utils/src/utils/support_framework/support_bundle.py (hunk 2 above)
Comment (vastradparayya) [reply py-utils/src/utils/support_framework/support_bundle.py]: Done

**Review by selvakumaar5496 — APPROVED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**